src/components/TextField.svelte

```diff
@@ -0,0 +1,116 @@
+<script lang="ts">
+    import { onMount } from 'svelte';
+    import { fade } from 'svelte/transition';
+
+    export let textFieldText: string;
+    export let errorCriteria: (val: string) => boolean = () => true; // Default to a function that always returns true
+    export let errorText = 'Error'; // Default error text
+
+    let value = '';
+    let isFocused = false;
+    let isValid = true;
+
+    const emptyState =
+        'w-602px h-68px font-suisse text-18px text-left bg-transparent border border-gray-600 text-white focus:outline-none transition duration-500 ease-in-out rounded-lg';
+    const focusedState =
+        'w-602px h-68px font-suisse text-18px text-left bg-transparent border border-purple-500 text-white focus:outline-none transition duration-500 ease-in-out rounded-lg';
+    const filledState =
+        'w-602px h-68px font-suisse text-18px text-left bg-transparent border border-red-500 text-white focus:outline-none transition duration-500 ease-in-out rounded-lg';
+
+    onMount(() => {
+        if (value) isFocused = true;
+    });
+
+    function handlePaste() {
+        navigator.clipboard.readText().then((clipText) => {
+            value = clipText;
+            isFocused = false;
+            isValid = errorCriteria(value);
+        });
+    }
+
+    function handleBlur() {
+        isFocused = false;
+        isValid = errorCriteria(value);
+    }
+</script>
+
+<div
+    class="relative h-68px w-602px absolute top-213px left-20px p-4 rounded-lg flex items-center gap-273px"
+>
+    <input
+        bind:value
+        type="text"
+        placeholder={textFieldText}
+        on:focus={() => (isFocused = true)}
+        on:blur={handleBlur}
+        class={value === ''
+            ? emptyState
+            : isFocused
+            ? focusedState
+            : isValid
+            ? emptyState
+            : filledState}
+        style="padding: 22.5px 100px 22.5px 20px;"
+    />
+    <button
+        on:click={handlePaste}
+        class="text-purple-600 font-semibold text-18px absolute right-10 top-1/2 transform -translate-y-1/2"
+        style="font-family: 'Open Sans', sans-serif; height: 25px; width: 48px;"
+    >
+        Paste
+    </button>
+    {#if !isFocused && !isValid && value != ''}
+        <p
+            transition:fade={{ duration: 500 }}
+            class="text-red-500 font-suisse text-18px leading-23px absolute bottom-0 transform translate-y-1/2"
+        >
+            {errorText}
+        </p>
+    {/if}
+</div>
+
+<style>
+    .border-gray-600 {
+        border-color: #545261;
+    }
+    .border-purple-500 {
+        border-color: #7162cf;
+    }
+    .border-red-500 {
+        border-color: #ef4444;
+    }
+    .h-60px {
+        height: 60px;
+    }
+    .h-100px {
+        height: 100px;
+    }
+    .w-602px {
+        width: 602px;
+    }
+    .top-213px {
+        top: 213px;
+    }
+    .left-20px {
+        left: 20px;
+    }
+    .gap-273px {
```

Comment:
Is there any room to compromise on these and go to the nearest Tailwind increment? `px` isn't necessarily a responsive unit of measure and these styles can't really be reused elsewhere.

Reply:
Agreed Micky has been pretty open about reworking designs to fit the tailwind standards. Might be worth updating to those units and just show him what it looks like. My guess is it'll be fine.